framework/include/samplers/Sampler.h

```diff
@@ -230,6 +239,9 @@ class Sampler : public MooseObject,
   void execute();
   friend void FEProblemBase::objectExecuteHelper<Sampler>(const std::vector<Sampler *> & objects);
 
+  LocalRankConfig _curr_rank_config;
+  bool _rank_config_set = false;
```

Comment:
`_rank_config_set` should be private. Otherwise, a Sampler will be able to changes its config anytime, which could be bad.

Reply:
It is private (unless I'm miss-reading the code).